### PR TITLE
Increase parser arrayLimit to 100

### LINKED ISSUE #5995: GraphQL: Selecting more than 20 items in `_in` filter results in ER_BAD_FIELD_ERROR
**Describe the bug**

**Steps to reproduce the behavior**

1. GraphQL needs to be installed
2. Create more than 20 of one content type
3. Execute a GraphQL Query that uses more than 20 items in `_in` filter. E.g.:
```
http://localhost:1337/graphql?query={documents(where:{id_in:[36,37,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59]}){id}}
```

**This is the error I get:** 
```
{
    "errors": [
        {
            "message": "select distinct `documents`.* from `documents` where `documents`.`id` in ({\"0\":\"36\",\"1\":\"37\",\"2\":\"40\",\"3\":\"41\",\"4\":\"42\",\"5\":\"43\",\"6\":\"44\",\"7\":\"45\",\"8\":\"46\",\"9\":\"47\",\"10\":\"48\",\"11\":\"49\",\"12\":\"50\",\"13\":\"51\",\"14\":\"52\",\"15\":\"53\",\"16\":\"54\",\"17\":\"55\",\"18\":\"56\",\"19\":\"57\",\"20\":\"58\",\"21\":\"59\"}) limit 10000 - ER_BAD_FIELD_ERROR: Unknown column '0' in 'where clause'",
            "locations": [
                {
                    "line": 1,
                    "column": 2
                }
            ],
            "path": [
                "documents"
            ],
            "extensions": {
                "code": "INTERNAL_SERVER_ERROR",
                "exception": {
                    "code": "ER_BAD_FIELD_ERROR",
                    "errno": 1054,
                    "sqlMessage": "Unknown column '0' in 'where clause'",
                    "sqlState": "42S22",
                    "index": 0,
                    "sql": "select distinct `documents`.* from `documents` where `documents`.`id` in (`0` = '36', `1` = '37', `2` = '40', `3` = '41', `4` = '42', `5` = '43', `6` = '44', `7` = '45', `8` = '46', `9` = '47', `10` = '48', `11` = '49', `12` = '50', `13` = '51', `14` = '52', `15` = '53', `16` = '54', `17` = '55', `18` = '56', `19` = '57', `20` = '58', `21` = '59') limit 10000"
                }
            }
        }
    ],
    "data": {
        "documents": null
    }
}
```

To me it looks like strapi or graphql want to convert it to an key-value object after 20 items.

strapi-graphql-distinct

With the Rest API I don't have that problem:
```
http://localhost:1337/documents?id_in=36&id_in=37&id_in=40&id_in=41&id_in=42&id_in=43&id_in=44&id_in=45&id_in=46&id_in=47&id_in=48&id_in=49&id_in=50&id_in=51&id_in=52&id_in=53&id_in=54&id_in=55&id_in=56&id_in=57&id_in=58&id_in=59
```

**Expected behavior**
Query should work as flawlessly as with <20 items
Also I don't know if `distinct` is correct here in the query.

**Screenshots**

I think Knex is doing some weird optimization and transforms a list of bindings longer than 20.
I run strapi with ENV `DEBUG=KNEX:*` and added console log for the bindings.

**With 20 items**
image

**With >20 items**
image

**System**

- Node.js version: v12.16.2
- NPM version: 6.14.4
- Strapi version: 3.0.0-beta.20.1
- Database: mysql (mariadb)
- Operating system: WSL/Linux

### PR DESCRIPTION
**Pull request by drj17**
#### Description of what you did:
Updated the arrayLimit in the parser to 100 (arbitrarily).  There might be a better number to use here, but it's causing issues when trying to do a where [field]_in query with a list of more than 20 items (see the tagged issue) using GraphQL.  You could theoretically use `Infinite` here but that might open things up to abuse. 

Fix #5995

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
@alexandrebodin is this something the user can just customize themselves within the middleware.js config file?

File: packages/strapi/lib/middlewares/parser/index.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const addQsParser = app => {
     get() {
       const qstr = this.querystring;
       const cache = (this._querycache = this._querycache || {});
-      return cache[qstr] || (cache[qstr] = qs.parse(qstr, { depth: 20 }));
+      return cache[qstr] || (cache[qstr] = qs.parse(qstr, { arrayLimit: 100, depth: 20 }));
```
Comment (alexandrebodin) [inline packages/strapi/lib/middlewares/parser/index.js:19]: I think we should alow passing some options so we can have a sensible default and leave the rest to the user. 

We can achieve this by adding a `queryStringParser` to `strapi.config.middleware.settings.parser` and setting the default in the default.json and passing the option to the `addQsParser` fn. We need to make sure we do not pass the option to koa-body though.

File: packages/strapi/lib/middlewares/parser/index.js [hunk 2]
```diff
@@ -2,21 +2,23 @@
...
     enumerable: true,
     /*
      * Get parsed query-string.
      */
     get() {
+      const { arrayLimit } = settings;
       const qstr = this.querystring;
       const cache = (this._querycache = this._querycache || {});
-      return cache[qstr] || (cache[qstr] = qs.parse(qstr, { depth: 20 }));
+      return cache[qstr] || (cache[qstr] = qs.parse(qstr, { arrayLimit, depth: 20 }));
```
Comment (alexandrebodin) [inline packages/strapi/lib/middlewares/parser/index.js:21]: can you move the depth to the default to and pass the entire setting object so a user can use extra params

File: packages/strapi/lib/middlewares/parser/index.js [hunk 3]
```diff
@@ -40,14 +42,13 @@ module.exports = strapi => {
...
-
         return body({
           patchKoa: true,
-          ...strapi.config.middleware.settings.parser,
+          ...omit(strapi.config.middleware.settings.parser, 'queryStringParser'),
         })(ctx, next);
       });
 
-      addQsParser(strapi.app);
+      addQsParser(strapi.app, strapi.config.middleware.settings.parser.queryStringParser);
```
Comment (alexandrebodin) [inline packages/strapi/lib/middlewares/parser/index.js:51]: can you replace the strapi.config.middleware.settings.parser.... access with  `strapi.config.get(...)`

**Comment by alexandrebodin** [comment]
@drj17 @derrickmehaffy can you add work together on adding this to the doc :)

**Comment by drj17** [comment]
Hi @derrickmehaffy, I took a first pass at adding it to the docs.  Let me know if there are any changes you'd like me to make!

**Comment by fellz** [comment]
release 3.1.4 please!  i am waiting for bug fix a week already....

**Comment by drj17** [comment]
> release 3.1.4 please! i am waiting for bug fix a week already....

[I posted a work-around here](https://github.com/strapi/strapi/issues/5995#issuecomment-672960802) if you need a quick fix

**Pull request merged**